script.js

```diff
@@ -1,4 +1,193 @@
+  var defaultMessage = secondMessage(playerNumberSecondStep);
+  return defaultMessage;
+};
+// kinda useless, i wanted to create a function that would make the main function as clean as possible
+var playerGameSequence = function (
+  diceGameSeq1,
+  diceGameSeq2,
+  dicePos,
+  playerNumberGameSeq
+) {
+  gameFirstStep(diceGameSeq1, diceGameSeq2, playerNumberGameSeq);
+  gameSecondStep(dicePos, playerNumberGameSeq);
+};
+//compare who's tens are bigger. return player value if their tens are bigger
+var isTensBigger = function () {
+  if (player1Array[0] > player2Array[0]) {
+    return 1;
+  } else if (player1Array[0] == player2Array[0]) {
+    return 0;
+  } else {
+    return 2;
+  }
+};
+//compare who's ones are bigger. return player value if their ones are bigger
+var isOnesBigger = function () {
+  if (player1Array[1] > player2Array[1]) {
+    return 1;
+  } else if (player1Array[1] == player2Array[1]) {
+    return 0;
+  } else {
+    return 2;
+  }
+};
+// to see which player has the bigger number
+var winCondition = function () {
+  // if p1 tens is equal to p2 tens
+  if (isTensBigger() == 0) {
+    //return isOnesBigger. We can deterine the winner/draw from isOnesBigger
+    return isOnesBigger();
+  } else {
+    //if tens is not equal, isTensBigger will return the winning player.
+    return isTensBigger();
+  }
+};
+// returns who wins or not
+var winConditionStatement = function () {
+  gameStep = 0;
+  var winningPlayer = winCondition();
+  if (winningPlayer != 0) {
+    return `Congrats Player ${winningPlayer}!! You have won! <br><br> Player 1's number is ${player1Array[0]}${player1Array[1]}. <br> Player 2's number is ${player2Array[0]}${player2Array[1]}. <br><br> Would you like to play again?`;
+  } else {
+    return `It's a draw! Would you like to play again?`;
+  }
+};
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  if (playerIsNumber == 1) {
+    if (gameStep == 0) {
+      mainDiceRoll1 = diceRoll();
+      mainDiceRoll2 = diceRoll();
+      myOutputValue = gameFirstStep(
+        mainDiceRoll1,
+        mainDiceRoll2,
+        playerIsNumber
+      );
+      gameStep = 1;
+    } else if (gameStep == 1) {
+      myOutputValue = gameSecondStep(
+        input,
+        playerIsNumber,
+        mainDiceRoll1,
+        mainDiceRoll2
+      );
+      playerIsNumber = 2;
+      gameStep = 0;
+    }
+  } else if (playerIsNumber == 2) {
+    if (gameStep == 0) {
+      mainDiceRoll1 = diceRoll();
+      mainDiceRoll2 = diceRoll();
+      myOutputValue = gameFirstStep(
+        mainDiceRoll1,
+        mainDiceRoll2,
+        playerIsNumber
+      );
+      gameStep = 1;
+    } else if (gameStep == 1) {
+      myOutputValue = gameSecondStep(
+        input,
+        playerIsNumber,
+        mainDiceRoll1,
+        mainDiceRoll2
+      );
+      playerIsNumber = 1;
+      gameStep = 2;
+    }
+  }
+  if (gameStep == 2) {
+    myOutputValue = winConditionStatement();
+  }
```

Comment:
These are pretty easy to read. good job with using functions to simplify the code in the main function👍